### PR TITLE
Optimization/early stopping

### PR DESCRIPTION
**Pull request by vindolineKis**
add early_stopping(patience,min_delta) into `trainerModel` ; add `optimizer.zero_grad()`to prevent gradient accumulation.

### CODE REVIEW

File: QCircLearning/trainerModel.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        if y is not None:
+            loss = self.loss_fn(pred, y)
+            return loss
+        else:
+            return pred
+
+    def __str__(self):
+        model_structure = self.model.__str__()
+        return f"TrainerModel(name={self.name}):\n{model_structure}"
+
```
Comment (cfenglv) [inline QCircLearning/trainerModel.py:30]: Suggestion: 
directly: "return f"TrainerModel(name={self.name}):\n{self.model}""
